### ISSUE TITLE
Askar-Anoncreds wallet does not work with multitenancy/askar-profile

### ISSUE THREAD

**Issue by ianco** [hyperledger/aries-cloudagent-python#2792]
For multitenancy, Askar supports two strategies for wallet storage:

- database-per-wallet - each wallet is stored in a separate database (this is the default)
- multi-wallet-per-database - multiple wallets are stored in a single database, using askar profiles

The second scenario is configured by providing the following options to aca-py:

```
--wallet-type askar
--multitenancy-config wallet_type=askar-profile
```

When using `--wallet-type askar-anoncreds`, aca-py throwns an error when trying to access the tenant wallet, one example is (from the faber demo):

```
Faber      |   File "/home/aries/aries_cloudagent/anoncreds/routes.py", line 196, in schemas_post
Faber      |     result = await issuer.create_and_register_schema(
Faber      |   File "/home/aries/aries_cloudagent/anoncreds/issuer.py", line 185, in create_and_register_schema
Faber      |     async with self.profile.session() as session:
Faber      |   File "/home/aries/aries_cloudagent/anoncreds/issuer.py", line 101, in profile
Faber      |     raise ValueError("AnonCreds interface requires AskarAnoncreds")
Faber      | ValueError: AnonCreds interface requires AskarAnoncreds
```

Looks like either a new profile is required under multitenancy, or the existing AskarProfile needs to be modified to support the `askar-anoncreds` scenario.

**Comment by swcurran** [comment]
Could this be related to #2789 and what it "undid" #2705?  Perhaps try it again now that #2789 has been merged and the change reverted?

**Comment by jamshale** [comment]
I don't think anoncreds fully supports multitenancy at this point. This is something i'm currently working on. 

Although `create_and_register_schema` was one endpoint that was supposed to be supported with default config. So this sounds like a separate issue other then just "not implemented yet". I can consider this use case when implementing multitenancy in general for the other endpoints.

**Comment by ianco** [comment]
> I don't think anoncreds fully supports multitenancy at this point. This is something i'm currently working on.

I think it's best to wait on the wallet upgrade (for multitenancy scenarios) until we're confident all the anoncreds/multitenancy is working correctly.  We could proceed tho with the "single wallet" upgrade scenario (which is another ticket).
